### PR TITLE
Configure the management Vault

### PR DESCRIPTION
**Pull request by iverberk**
This PR adds Terraform code to configure the management Vault with authentication methods and secret engines.

### CODE REVIEW

File: management/infra/ad.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # Providing the owners is optional as this also can be managed from the Azure Portal
 
 resource "azuread_group" "platform_admin" {
-  display_name            = "platform-admin"
+  display_name            = "platform-admin-new"
```
Comment (arminc) [inline management/infra/ad.tf:7]: remove -new

File: management/infra/ad.tf [hunk 2]
```diff
@@ -13,8 +13,8 @@ resource "azuread_group" "platform_admin" {
   owners = var.management_infra_platform_admin_owners
 }
 
-resource "azuread_group" "platform_developer" {
-  display_name            = "platform-developer"
+resource "azuread_group" "platform_operator" {
+  display_name            = "platform-operator-new"
```
Comment (arminc) [inline management/infra/ad.tf:17]: here as well

File: management/infra/ad.tf [hunk 3]
```diff
@@ -23,7 +23,7 @@ resource "azuread_group" "platform_developer" {
 }
 
 resource "azuread_group" "platform_viewer" {
-  display_name            = "platform-viewer"
+  display_name            = "platform-viewer-new"
```
Comment (arminc) [inline management/infra/ad.tf:26]: here

File: management/infra/terraform.tf [hunk 4]
```diff
@@ -29,7 +29,6 @@ terraform {
 
   backend "azurerm" {
     resource_group_name  = "terraform-state"
-    storage_account_name = "This name should match management_bootstrap_terraform_state_account_name"
```
Comment (arminc) [inline management/infra/terraform.tf:32]: why is this removed?

File: management/vault/config/secret-engine.tf [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  max_lease_ttl_seconds     = 14400
+}
+
+resource "vault_aws_secret_backend_role" "aws_owner_role" {
+  for_each = { for account in var.management_vault_config_accounts : account.name => account }
+
+  backend         = vault_aws_secret_backend.aws[each.key].path
+  name            = "vault-aws-secrets"
+  credential_type = "assumed_role"
+  role_arns       = ["arn:aws:iam::${each.value.id}:role/vault-aws-secrets"] # This role will be configured with AWS organizations
```
Comment (arminc) [inline management/vault/config/secret-engine.tf:106]: Is -secrets suffix explicit enough? Maybe we should call it vault-aws-access or something like that?

Comment (iverberk) [reply management/vault/config/secret-engine.tf:106]: It's analogous to the Azure service principal name. I'm okay with renaming it to vault-aws-access and vault-azure-access

File: management/infra/terraform.tf (hunk 4 above)
Comment (iverberk) [reply management/infra/terraform.tf:32]: Because we don't want people to make adjustments in the terraform.tf file? This should be passed on the CLI as a variable to prevent hardcoding of the value here and having a permanent change in the repo that is only for local use.
